process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java

```diff
@@ -0,0 +1,58 @@
+package org.kie.processmigration.service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.persistence.EntityManager;
+import javax.persistence.Persistence;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+@Startup
+@ApplicationScoped
+public class EntityManagerProducer {
```

Comment:
Has this class been added only for the tests requirements?

Reply:
No - Since we want to allow customers to either auto-generate PIM schema through Hibernate or create it based on the DDL scripts provided, we need to set the Hibernate properties (`hibernate.hbm2ddl.auto` in this case) dynamically. Thus, if you set these properties as a system properties, they will be taken into account when creating the entity manager factory and overwrite the ones defined in the persistence.xml file

See: 
https://github.com/kiegroup/droolsjbpm-integration/blob/ba5039e6683c2faac2ae632c03bf29506e260df0/process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java#L56